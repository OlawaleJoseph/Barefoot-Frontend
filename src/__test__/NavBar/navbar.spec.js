import { shallow } from 'enzyme';
import React from 'react';
import Navbar from '../../components/navbar';
import Login from '../../components/login';

describe('Landing Page Navbar', () => {
  const wrapper = shallow(<Navbar />);
  describe('Brand', () => {
    it('Should render properly', () => {
      const brand = wrapper.find('.navbar-brand');
      expect(brand).toHaveLength(1);
      expect(brand.text()).toEqual(' BareFoot ');
    });
  });

  describe('Navigation Links', () => {
    const navLinks = wrapper.find('.nav-link');
    it('Should render properly', () => {
      const navList = wrapper.find('.navbar-collapse');
      expect(navList).toHaveLength(1);
      expect(navLinks).toHaveLength(4);
    });
    it('Should render Login Button', () => {
      const loginBtn = wrapper.find('.login-btn');
      const login = shallow(<Login />);
      expect(loginBtn).toHaveLength(1);
      expect(loginBtn.text()).toEqual('LOGIN');
      expect(login).toMatchSnapshot();
    });
  });
});
