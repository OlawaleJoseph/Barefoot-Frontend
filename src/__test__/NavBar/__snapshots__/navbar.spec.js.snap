// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landing Page Navbar Navigation Links Login Button 1`] = `ShallowWrapper {}`;
